src/Differ/MapDiffer.php

```diff
@@ -159,16 +159,10 @@ private function getDiffForArrays( array $old, array $new ) {
 	 *
 	 * @param array $array
 	 *
-	 * @return boolean
+	 * @return bool
 	 */
 	private function isAssociative( array $array ) {
-		foreach ( $array as $key => $value ) {
-			if ( is_string( $key ) ) {
-				return true;
-			}
-		}
-
-		return false;
+		return (bool)array_filter( array_keys( $array ), 'is_string' );
```

Comment:
So if you got an associative array with a thousand elements, it'll now do a thousand iterations instead of one? Decreasing the quality of the code just to match a fault CS rule is the opposite of what these tools intend you to do.